File: src/main/java/com/example/pidev/model/budget/Budget.java
package com.example.pidev.model.budget;

public class Budget {
    private int id;
    private int event_id;
    private double initial_budget;
    private double total_expenses;
    private double total_revenue;
    private double rentabilite;

    public Budget() {}

    public Budget(int id, int event_id, double initial_budget, double total_expenses, double total_revenue, double rentabilite) {
        this.id = id;
        this.event_id = event_id;
        this.initial_budget = initial_budget;
        this.total_expenses = total_expenses;
        this.total_revenue = total_revenue;
        this.rentabilite = rentabilite;
    }

    public Budget(int event_id, double initial_budget, double total_revenue) {
        this.event_id = event_id;
        this.initial_budget = initial_budget;
        this.total_revenue = total_revenue;
        this.total_expenses = 0;
        this.rentabilite = total_revenue;
    }

    public int getId() { return id; }
    public int getEvent_id() { return event_id; }
    public double getInitial_budget() { return initial_budget; }
    public double getTotal_expenses() { return total_expenses; }
    public double getTotal_revenue() { return total_revenue; }
    public double getRentabilite() { return rentabilite; }

    public void setId(int id) { this.id = id; }
    public void setEvent_id(int event_id) { this.event_id = event_id; }
    public void setInitial_budget(double initial_budget) { this.initial_budget = initial_budget; }
    public void setTotal_expenses(double total_expenses) { this.total_expenses = total_expenses; }
    public void setTotal_revenue(double total_revenue) { this.total_revenue = total_revenue; }
    public void setRentabilite(double rentabilite) { this.rentabilite = rentabilite; }
}

File: src/Entity/Budget/Budget.php
<?php

namespace App\Entity\Budget;

use App\Repository\Budget\BudgetRepository;
use Doctrine\DBAL\Types\Types;
use Doctrine\ORM\Mapping as ORM;
use Symfony\Component\Validator\Constraints as Assert;

#[ORM\Entity(repositoryClass: BudgetRepository::class)]
#[ORM\Table(name: 'budget')]
class Budget
{
    #[ORM\Id]
    #[ORM\GeneratedValue(strategy: 'IDENTITY')]
    #[ORM\Column(type: Types::INTEGER)]
    private ?int $id = null;

    #[ORM\Column(name: 'event_id', type: Types::INTEGER)]
    #[Assert\Positive(message: 'L\'evenement est obligatoire.')]
    private ?int $eventId = null;

    #[ORM\Column(name: 'initial_budget', type: Types::DECIMAL, precision: 10, scale: 2)]
    #[Assert\NotBlank(message: 'Le budget initial est obligatoire.')]
    #[Assert\PositiveOrZero(message: 'Le budget initial doit etre positif.')]
    private string $initialBudget = '0.00';

    #[ORM\Column(name: 'total_expenses', type: Types::DECIMAL, precision: 10, scale: 2, options: ['default' => '0.00'])]
    #[Assert\PositiveOrZero(message: 'Les depenses doivent etre positives.')]
    private string $totalExpenses = '0.00';

    #[ORM\Column(name: 'total_revenue', type: Types::DECIMAL, precision: 10, scale: 2, options: ['default' => '0.00'])]
    #[Assert\PositiveOrZero(message: 'Les revenus doivent etre positifs.')]
    private string $totalRevenue = '0.00';

    #[ORM\Column(name: 'rentabilite', type: Types::DECIMAL, precision: 10, scale: 2, options: ['default' => '0.00'])]
    private string $rentabilite = '0.00';

    public function getId(): ?int
    {
        return $this->id;
    }

    public function getEventId(): ?int
    {
        return $this->eventId;
    }

    public function setEventId(?int $eventId): self
    {
        $this->eventId = $eventId;

        return $this;
    }

    public function getInitialBudget(): string
    {
        return $this->initialBudget;
    }

    public function setInitialBudget(float|string $initialBudget): self
    {
        $this->initialBudget = number_format((float) $initialBudget, 2, '.', '');

        return $this;
    }

    public function getTotalExpenses(): string
    {
        return $this->totalExpenses;
    }

    public function setTotalExpenses(float|string $totalExpenses): self
    {
        $this->totalExpenses = number_format((float) $totalExpenses, 2, '.', '');

        return $this;
    }

    public function getTotalRevenue(): string
    {
        return $this->totalRevenue;
    }

    public function setTotalRevenue(float|string $totalRevenue): self
    {
        $this->totalRevenue = number_format((float) $totalRevenue, 2, '.', '');

        return $this;
    }

    public function getRentabilite(): string
    {
        return $this->rentabilite;
    }

    public function setRentabilite(float|string $rentabilite): self
    {
        $this->rentabilite = number_format((float) $rentabilite, 2, '.', '');

        return $this;
    }

    public function refreshRentabilite(): self
    {
        $this->setRentabilite((float) $this->totalRevenue - (float) $this->totalExpenses);

        return $this;
    }
}




